File: year2021/python/day2/day2.py
from year2021.python.day2.day2_func import *

moves = [Move(move) for move in open('../../data/day2_data.txt')]

calculator = RouteCalculator()
position = calculator.GetPosition(moves)

print(f"Part 1: {position}")

aimCalculator = AimCalculator()
aimPosition = aimCalculator.GetPosition(moves)

print(f"Part 2: {aimPosition}")

File: year2021/c++/day2/main/day2.cpp
#include "../include/day2_func.h"
#include <fstream>
#include <iostream>

using namespace std;

int main() {
    ifstream ifs("../../../data/day2_data.txt");
    string line;
    vector<Move> moves;
    while(getline(ifs, line)) {
        moves.push_back(Move(line));
    }

    RouteCalculator routeCalculator;
    int positiion = routeCalculator.GetPosition(moves);

    cout << "Part 1: " << positiion << endl;

    AimCalculator aimCalculator;
    int aimPosition = aimCalculator.GetPosition(moves);

    cout << "Part 2: " << aimPosition << endl;
}